### PR TITLE
Testing: add git to PATH

### LINKED ISSUE #303: Testing: fixture to temporarily add certain utilities to binary path
**How would this feature be useful?**

Currently, with a package that requires compilation, tests that install it fail on macOS (such as regex dependency inside of black).

Additionally, it would be helpful to add `git` and other VCS tools so that we could test installing package specifications with VCS urls.

**Describe the solution you'd like**

The solution that @chrahunt mentioned (https://github.com/pipxproject/pipx/pull/298#issuecomment-568642820), quoted below:

1. Have a session-scoped fixture which creates a temporary directory
1. Populate the directory with symlinks to whitelisted build tools identified by shutil.which like gcc/clang
1. Depend on that fixture here, adding the temporary directory to PATH

**Describe alternatives you've considered**
Previously, we just added some paths to the bin path (see PR #269 to fix Issue #266) but this resulted in undesired side-effects and was removed (#298, #299 )

### PR DESCRIPTION
**Pull request by dukecat0**
* [ ] I have added an entry to `docs/changelog.md`

## Summary of changes

Add git to PATH during testing, so we can test installing packages with VCS urls.
Closes #303 

## Test plan

Tested by running
```
nox -s tests
```

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -139,14 +139,15 @@ def pipx_session_shared_dir(tmp_path_factory):
...
     tmp_path = tmp_path_factory.mktemp("session_utilstempdir")
-    try:
-        Path(tmp_path / "git.exe" if WIN else tmp_path / "git").symlink_to(
-            Path(git_path)
-        )
-    except FileExistsError:
-        pass
+    utils = ["git"]
+    for util in utils:
+        util_filename = f"{util}.exe" if WIN else util
```
Comment (uranusjr) [inline tests/conftest.py:145]: Not needed, `shutil.which` do this automatically.

File: tests/conftest.py [hunk 2]
```diff
@@ -59,7 +62,7 @@ def pipx_temp_env_helper(pipx_shared_dir, tmp_path, monkeypatch, request):
     #   which make tests fail (e.g. on Github ansible apps exist in /usr/bin)
     monkeypatch.setenv("PATH_ORIG", str(bin_dir) + os.pathsep + os.getenv("PATH"))
     monkeypatch.setenv("PATH_TEST", str(bin_dir))
-    monkeypatch.setenv("PATH", str(bin_dir))
+    monkeypatch.setenv("PATH", str(bin_dir) + os.pathsep + str(utils_temp_dir))
```
Comment (uranusjr) [inline tests/conftest.py:65]:
```suggestion
    monkeypatch.setenv("PATH", str(bin_dir / utils_temp_dir))
```

File: tests/conftest.py [hunk 3]
```diff
@@ -134,8 +137,23 @@ def pipx_session_shared_dir(tmp_path_factory):
...
 
 
+@pytest.fixture(scope="session")
+def utils_temp_dir(tmp_path_factory):
+    tmp_path = tmp_path_factory.mktemp("session_utilstempdir")
+    utils = ["git"]
+    for util in utils:
+        util_path = Path(shutil.which(util))
+        try:
+            Path(tmp_path / util_path.name).symlink_to(util_path)
```
Comment (uranusjr) [inline tests/conftest.py:147]:
```suggestion
           (tmp_path / util_path.name).symlink_to(util_path)
```

File: tests/conftest.py (hunk 2 above)
Comment (dukecat0) [reply tests/conftest.py:65]: Seems like this will make the tests fail?

Comment (uranusjr) [reply tests/conftest.py:65]: Why? But eh this ugly construct is also used by existing code so maybe it’s better to be consistent and do the refactoring in some other PR.

Comment (dukecat0) [reply tests/conftest.py:65]: The `bin_dir` is the location for apps installed by `pipx install`, so these two directories should be separated.

**Review by uranusjr — APPROVED** [review]

**Pull request merged**